### PR TITLE
[MRG+1]: Allow Montage to set info['dig']

### PR DESCRIPTION
**Pull request by larsoner**
This PR:

1. unifies how the `Montage` class exposes the fiducials
2. allows `set_montage` to set `info['dig']` for the `Montage` class, which is useful e.g. for `mne.coreg` and `fit_sphere_to_headshape`.

It's not the full overhaul we need to deal with Montage/DigMontage, but it helps some (and doesn't prevent the eventual full refactoring).

### CODE REVIEW

File: mne/channels/tests/test_montage.py [hunk 1]
```diff
@@ -187,16 +187,14 @@ def test_montage():
     Fp1 -95.0 -31.0 -3.0
     AF7 -81 -59 -3
     AF3 -87 -41 28
-    nasion -91 0 -42
-    lpa 0 -91 -42
-    rpa 0 91 -42
+    FidNz -91 0 -42
+    FidT9 0 -91 -42
+    FidT10 0 91 -42
```
Comment (agramfort) [inline mne/channels/tests/test_montage.py]: these names are more standard? can you clarify why this change?

**Comment by agramfort** [comment]
why would you want to call set_montage with set_dig=False ?

**Comment by larsoner** [comment]
Not sure, mostly have it for backward compatibility just in case but
perhaps it's not necessary

File: mne/channels/tests/test_montage.py (hunk 1 above)
Comment (larsoner) [inline mne/channels/tests/test_montage.py]: All of our sfp files in channels/data/montages have these as the fiducials

Comment (larsoner) [inline mne/channels/tests/test_montage.py]: Looks like this is expected, too

http://wiki.besa.de/index.php?title=Electrodes_and_Surface_Locations

**Comment by agramfort** [comment]
so this changes the current behavior already?

**Comment by agramfort** [comment]
hum ok...no idea here

**Comment by larsoner** [comment]
> so this changes the current behavior already

Yeah, but in a way that shouldn't really shouldn't break people's code (but
the option is there to get the old behavior just in case I'm wrong)

**Comment by larsoner** [comment]
Comment added about SFP fiducials

**Comment by agramfort** [comment]
+1 for MRG

**Comment by larsoner** [comment]
@mmagnuski I saw some :+1: from you are you happy here, too?

**Comment by mmagnuski** [comment]
Initial 👍 was because I like the idea, but now I looked at the changes and I am +1 for MRG too.

**Comment by larsoner** [comment]
Thanks for the quick looks
